lib/experimental/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -141,12 +141,10 @@ public static function get_block_data() {
 		 *
 		 * @param WP_Theme_JSON_Data_Gutenberg Class to access and update the underlying data.
 		 */
-		$theme_json = apply_filters( 'theme_json_blocks', new WP_Theme_JSON_Data_Gutenberg( $config, 'core' ) );
```

Comment:
By using an unregistered origin this is what was happening:

- The `WP_Theme_JSON_Resolver` loaded the blocks origin (named `core`).
- In the `WP_Theme_JSON` constructor, because it was [not one of the origins registered ](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/class-wp-theme-json.php#L507) it'll be changed to `theme`.
- The `WP_Theme_JSON_Resolved` loaded then the `theme` origin, updating any instance.

This was working as expected. But only by coincidence. If the blocks origin was able to define presets this won't work because the presets would be stored in the `theme` key and when the actual theme origin loaded would override them.